scripts/test_screeps_rl_simulator_harness.py

```
@@ -6129,6 +6129,125 @@ def test_failed_mongo_room_summary_exposes_collection_error(self) -> None:
+        self.assertEqual(token, "placed-token")
+        self.assertEqual(smoke.calls, 2)
+        sleep.assert_called_once_with(0.25)
+        debug_phase.assert_called_once()
+        self.assertEqual(summary["classification"], "ok")
+        self.assertTrue(summary["retry"]["recovered"])
+        self.assertEqual(summary["retry"]["attempts"][0]["classification"], "room_busy")
+
+    def test_place_spawn_retry_reports_persistent_room_busy_with_no_rerun_guidance(self) -> None:
+        class Result:
+            status = 200
+            payload = {"error": "room busy"}
```

Ruff (0.15.14)

[warning] Mutable default value for class attribute

(RUF012)